sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinitionXmlLoader.java

```diff
@@ -159,76 +227,117 @@ private void processRule(RulesDefinition.NewRepository repo, SMInputCursor ruleC
-    String status = null;
-    Cardinality cardinality = Cardinality.SINGLE;
+    RuleStatus status = RuleStatus.defaultStatus();
+    boolean template = false;
+    String effortToFixDescription = null;
+    String debtSubCharacteristic = null;
+    String debtRemediationFunction = null;
+    String debtRemediationFunctionOffset = null;
+    String debtRemediationFunctionCoeff = null;
     List<ParamStruct> params = new ArrayList<>();
     List<String> tags = new ArrayList<>();
 
     /* BACKWARD COMPATIBILITY WITH VERY OLD FORMAT */
     String keyAttribute = ruleC.getAttrValue("key");
-    if (StringUtils.isNotBlank(keyAttribute)) {
-      key = StringUtils.trim(keyAttribute);
+    if (isNotBlank(keyAttribute)) {
+      key = trim(keyAttribute);
     }
     String priorityAttribute = ruleC.getAttrValue("priority");
-    if (StringUtils.isNotBlank(priorityAttribute)) {
-      severity = StringUtils.trim(priorityAttribute);
+    if (isNotBlank(priorityAttribute)) {
+      severity = trim(priorityAttribute);
     }
 
     SMInputCursor cursor = ruleC.childElementCursor();
     while (cursor.getNext() != null) {
       String nodeName = cursor.getLocalName();
 
-      if (StringUtils.equalsIgnoreCase("name", nodeName)) {
-        name = StringUtils.trim(cursor.collectDescendantText(false));
+      if (equalsIgnoreCase("name", nodeName)) {
+        name = trim(cursor.collectDescendantText(false));
 
-      } else if (StringUtils.equalsIgnoreCase("description", nodeName)) {
-        description = StringUtils.trim(cursor.collectDescendantText(false));
+      } else if (equalsIgnoreCase("description", nodeName)) {
+        description = trim(cursor.collectDescendantText(false));
 
-      } else if (StringUtils.equalsIgnoreCase("key", nodeName)) {
-        key = StringUtils.trim(cursor.collectDescendantText(false));
+      } else if (equalsIgnoreCase("key", nodeName)) {
+        key = trim(cursor.collectDescendantText(false));
 
-      } else if (StringUtils.equalsIgnoreCase("configKey", nodeName)) {
+      } else if (equalsIgnoreCase("configKey", nodeName)) {
         // deprecated field, replaced by internalKey
-        internalKey = StringUtils.trim(cursor.collectDescendantText(false));
+        internalKey = trim(cursor.collectDescendantText(false));
 
-      } else if (StringUtils.equalsIgnoreCase("internalKey", nodeName)) {
-        internalKey = StringUtils.trim(cursor.collectDescendantText(false));
+      } else if (equalsIgnoreCase("internalKey", nodeName)) {
+        internalKey = trim(cursor.collectDescendantText(false));
 
-      } else if (StringUtils.equalsIgnoreCase("priority", nodeName)) {
+      } else if (equalsIgnoreCase("priority", nodeName)) {
         // deprecated field, replaced by severity
-        severity = StringUtils.trim(cursor.collectDescendantText(false));
+        severity = trim(cursor.collectDescendantText(false));
+
+      } else if (equalsIgnoreCase("severity", nodeName)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 270. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)